### PR TITLE
[#666] Extended Gradle build for JUnit 5 test execution

### PR DESCRIPTION
**Pull request by kthoms**
Adds a dependency on org.junit.jupiter:junit-jupiter-engine:5.1.0.

### CODE REVIEW

File: org.eclipse.xtext.xtext.wizard/src/org/eclipse/xtext/xtext/wizard/TestProjectDescriptor.xtend [hunk 1]
```diff
@@ -64,6 +64,16 @@ abstract class TestProjectDescriptor extends ProjectDescriptor {
...
 				maven.scope = Scope.TESTCOMPILE
 			]
+			if (config.needsGradleBuild) {
+				deps += new ExternalDependency()=>[
+					p2.bundleId = "org.junit.jupiter.engine"
+					p2.version = "5.1.0"
+					maven.groupId = "org.junit.jupiter"
+					maven.artifactId = "junit-jupiter-engine"
+					maven.version = "5.1.0"
+					maven.scope = Scope.TESTCOMPILE
```
Comment (cdietrich) [inline org.eclipse.xtext.xtext.wizard/src/org/eclipse/xtext/xtext/wizard/TestProjectDescriptor.xtend]: in testRuntime in gradle junit5 example

**Comment by cdietrich** [comment]
does not work in gradle/maven layout

**Comment by kthoms** [comment]
Introduced TESTRUNTIME scope

**Comment by kthoms** [comment]
@cdietrich please review, should work now

**Comment by cdietrich** [comment]
as i said: does not work with maven layout

**Comment by cdietrich** [comment]
(no test block in runtime project)

**Pull request merged**